### ISSUE TITLE
Improve highlights downloading: add token for highlight title for dirname/filename_pattern; download only highlights matching a title

### ISSUE THREAD

**Issue by EveryDirection** [instaloader/instaloader#1684]
Currently, for the parameters `--dirname-pattern` and `--filename-pattern`, there is no available token, such as a hypothetical `{highlight_title}`, for the highlight title.

When downloading highlights with the `--highlights` parameter, there is thus no way to organize the downloaded files by which highlight they belonged to, and all the files just end up an undifferentiated mass in the same download directory, instead of being organized by highlight title as they are on the Instagram profile. And if we afterwards manually organize the downloaded files into different directories, without something like `--filename-pattern={highlight_title}\...`, on a download session later in the future, there is no way for Instaloader to check/know if a highlight file has already been downloaded on a previous session.

Such an available token would also be useful for downloading only highlights that match a specific title; in this example, we are matching for the title "_foobar_":

`--highlights-filter="highlight_title==foobar"`

**Comment by uvx0ol** [comment]
> there is no way for Instaloader to check/know if a highlight file has already been downloaded on a previous session.

I realized Instaloader will always check all the highlights on an account you're trying to download them from so it also creates dozens of unnecessary requests. It makes no sense as probably nobody uploads anything new to these old highlights.

**Comment by EveryDirection** [comment]
> No. Downloading with `--highlights` will set the target to profile/highlight_title. Thus with the default `--dirname-pattern={target}`, Instaloader will place each highlight into a different folder, opposed to how you present it here.

Hi, I very much appreciate your reply, as I learned something new and was not previously aware of this behavior for {target}.

I tested it, and the behavior is indeed as you described. I didn't know about this behavior because it is different from that described by the documentation at https://instaloader.github.io/cli-options.html, which says:

> --dirname-pattern DIRNAME_PATTERN

> {target} is replaced by the target you specified, i.e. either :feed, #hashtag or the profile name.

This shows what I am talking about: there should be a separate token for highlight title. {target} should refer to just the target specified, not a combination of the target with something else.

For example, if you are downloading with...:

`instaloader --dirname-pattern={target} --highlights --no-posts dogsofinstagram`

...then the target is "dogsofinstagram". So the documentation would suggest that the highlights would be downloaded to the directory pattern "dogsofinstagram", but instead, the highlights are downloaded to the directory pattern "dogsofinstagram/highlight_title".

Also, what if we want to download do a directory that is not in that exact format, for example, to the directory pattern "foobar/highlight_title" or to the directory pattern "dogsofinstagram/Highlights/highlight_title"? Currently, there is no way to do that, since `--dirname-pattern={target}` forces the directory pattern "dogsofinstagram/highlight_title".

`{highlight_title}` should be a separate token from `{target}`, so that we can, for example, download to "foobar/highlight_title" by setting `--dirname-pattern=foobar/{highlight_title}` or download to "dogsofinstagram/Highlights/highlight_title" by setting `--dirname-pattern={target}/{highlight_title}`.
